superset/utils/core.py

```diff
@@ -178,18 +181,21 @@ def string_to_num(s: str):
         return None
 
 
-class DimSelector(Having):
-    def __init__(self, **args):
-        # Just a hack to prevent any exceptions
-        Having.__init__(self, type="equalTo", aggregation=None, value=None)
+try:
+    class DimSelector(Having):
+        def __init__(self, **args):
+            # Just a hack to prevent any exceptions
+            Having.__init__(self, type='equalTo', aggregation=None, value=None)
 
-        self.having = {
-            "having": {
-                "type": "dimSelector",
-                "dimension": args["dimension"],
-                "value": args["value"],
+            self.having = {
+                'having': {
+                    'type': 'dimSelector',
+                    'dimension': args['dimension'],
+                    'value': args['value'],
+                },
             }
-        }
+except NameError:
+    pass
```

Comment:
For clarity, it might be better to have this try-except block done together with the import, i.e. first trying to import `Having`, and if that's found, continue with defining `DimSelector`. Right now they are quite far apart, and makes understanding the optionality slightly less obvious.

Reply:
I've hoisted the class declaration and put it first thing after the imports. In theory, it's doable to also declare the class inside the `try` block, but I'm afraid of flake. Or maybe you meant something else?